kim_convergence/ase/equilibration.py

```
@@ -0,0 +1,232 @@
+        >>> result = run_ase_equilibration(sampler, relative_accuracy=0.1)
+    """
+    # Build kwargs for run_length_control
+    # Prevent override of reserved parameters that are critical to function operation
+    reserved_keys = {"get_trajectory", "fp", "fp_format"}
+    conflicts = reserved_keys.intersection(kwargs)
+    if conflicts:
+        raise ValueError(
+            f"Cannot override reserved parameter(s): {', '.join(sorted(conflicts))}. "
+            "'get_trajectory' is automatically set from the provided sampler, "
+            "and 'fp'/'fp_format' are required for result parsing."
+        )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)